plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -312,13 +307,7 @@ public IcebergTableHandle getTableHandle(
             return null;
         }
 
-        if (name.getSnapshotId().isPresent() && endVersion.isPresent()) {
-            throw new TrinoException(GENERIC_USER_ERROR, "Cannot specify end version both in table name and FOR clause");
-        }
-
-        Optional<Long> snapshotId = endVersion.map(version -> getSnapshotIdFromVersion(table, version))
-                .or(() -> getSnapshotId(table, name.getSnapshotId(), isAllowLegacySnapshotSyntax(session)));
```

Comment:
This used to be doing `Optional.ofNullable(table.currentSnapshot()).map(Snapshot::snapshotId)`.
Was the removal intentional?

Reply:
We need that

Reply:
Appologies. The removal was not intentional.